### PR TITLE
Use NodeId type from client crate

### PR DESCRIPTION
**Pull request by Wiezzel**
Replace all usage of `ya_core_model::ethaddr::NodeId` with `ya_client_model::NodeId`.

### CODE REVIEW

File: core/market-forwarding/Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ edition = "2018"
 
 [dependencies]
 ya-client = "0.3"
+ya-client-model = "0.1"
```
Comment (tworec) [inline core/market-forwarding/Cargo.toml:10]: not required, since it is visible via `ya_client::model`

File: core/market-forwarding/src/api.rs [hunk 2]
```diff
@@ -3,7 +3,7 @@ use jsonwebtoken::{encode, Header};
 use serde::{Deserialize, Serialize};
 use url::Url;
 
-use ya_core_model::ethaddr::NodeId;
+use ya_client_model::NodeId;
```
Comment (tworec) [inline core/market-forwarding/src/api.rs:6]:
```suggestion
use ya_client::model::NodeId;
```

File: core/market-forwarding/Cargo.toml (hunk 1 above)
Comment (Wiezzel) [reply core/market-forwarding/Cargo.toml:10]: OK, I will update this for all crates.

File: core/payment/Cargo.toml [hunk 3]
```diff
@@ -10,7 +10,7 @@ gnt-driver = []
 dummy-driver = []
 
 [dependencies]
-ya-core-model = { version = "0.1", features = ["payment", "market", "identity", "with-diesel"] }
+ya-core-model = { version = "0.1", features = ["payment", "market", "identity"] }
 ya-client-model = "0.1"
```
Comment (tworec) [inline core/payment/Cargo.toml:14]: don't you need?
```suggestion
ya-client-model = { version = "0.1", features = [ "with-diesel" ] }
```

Comment (Wiezzel) [reply core/payment/Cargo.toml:14]: Looks like it's needed, but the compiler doesn't complain... :thinking:

File: core/payment/src/service.rs [hunk 4]
```diff
@@ -146,7 +146,7 @@ mod public {
         };
 
         let offeror_id = agreement.offer.provider_id.clone().unwrap(); // FIXME: provider_id shouldn't be an Option
-        let issuer_id = debit_note.issuer_id.clone();
+        let issuer_id = debit_note.issuer_id.to_string();
```
Comment (tworec) [inline core/payment/src/service.rs:149]: id rather parse sender_id here and compare node_ids below

File: core/payment/Cargo.toml (hunk 3 above)
Comment (tworec) [reply core/payment/Cargo.toml:14]: try to compile from within the crate directory... ;)

File: core/persistence/Cargo.toml [hunk 5]
```diff
@@ -6,7 +6,7 @@ edition = "2018"
 
 
 [dependencies]
-ya-core-model = { version = "0.1", features=["ethaddr"]}
+ya-core-model = "0.1"
 ya-client-model = "0.1"
```
Comment (tworec) [inline core/persistence/Cargo.toml:10]:
```suggestion
ya-client-model = { version = "0.1", features = [ "with-diesel" ] }
```

File: core/payment/src/service.rs (hunk 4 above)
Comment (Wiezzel) [reply core/payment/src/service.rs:149]: But parsing requires error handling and this doesn't. Also, I would have to parse offeror_id as as well. So how is that better?

Comment (tworec) [reply core/payment/src/service.rs:149]: it was a suggestion / non-blocking comment :)

File: core/payment/Cargo.toml (hunk 3 above)
Comment (Wiezzel) [reply core/payment/Cargo.toml:14]: Oh, right. I think Cargo should detect this nevertheless. If building of any of the crates alone fails, so should building the whole workspace.

File: core/payment/src/service.rs (hunk 4 above)
Comment (Wiezzel) [reply core/payment/src/service.rs:149]: OK :)

**Pull request merged**